assistance/Workflow.py

```diff
@@ -5,19 +5,93 @@
 Description: Workflow class provides all the information needed for the filtering of workflows in assistance manual.
 
 """
+from utils.WebTools import getResponse
+from utils.SearchTools import findKeys
 
 class Workflow:
 
-    def _init_(self, workflow, url='cmsweb.cern.ch'):
+    def __init__(self, workflowName, url='cmsweb.cern.ch'):
         """
         Initialize the Workflow class
-        :param str workflow: is the name of the workflow
+        :param str workflowName: is the name of the workflow
         :param str url: is the url to fetch information from
         """
 
-        super(Workflow, self)._init_()
-        self.workflow = workflow
+        self.workflowName = workflowName
         self.url = url
+        self.workflowParams = self.getWorkflowParams()
+
+    def getWorkflowParams(self):
+
+        """
+        Get the workflow parameters from ReqMgr2.
+        See the `ReqMgr 2 wiki <https://github.com/dmwm/WMCore/wiki/reqmgr2-apis>`_
+        for more details.
+        :returns: Parameters for the workflow from ReqMgr2.
+        :rtype: dict
+        """
+
+        try:
+            result = getResponse(url=self.url,
+                                 endpoint='/reqmgr2/data/request/',
+                                 param=self.workflowName)
+
+            for params in result['result']:
+                for key, item in params.items():
+                    if key == self.workflowName:
+                        self.workflowParams = item
+                        return item
+
+        except Exception as error:
+            print 'Failed to get workflow params from reqmgr for %s '% self.workflowName
+            print str(error) 
+
+    def getPrepID(self):
+        """
+        :param: None
+        :returns: PrepID
+        :rtype: string
+        """
+        return self.workflowParams.get('PrepID')
```

Comment:
Requests can have multiple PrepID. Does it matter in your case?

Reply:
Really? I did not know that. I thought it's unique for a request and I haven't seen any occurrence of multiple PrepIDs. What's use of having multiple of them? Do you have an example at hand?

Reply:
Sure, there must be plenty in the system, here is one:
https://cmsweb.cern.ch/reqmgr2/fetch?rid=pdmvserv_task_HIG-RunIISummer20UL16wmLHEGENAPV-00196__v1_T_210224_212517_8184

PrepID is an identifier needed by offline/McM, since multiple "McM requests" can be created as a single "ReqMgr2 request", that's how they keep track where each PrepID request is.

Reply:
Ah, I got it. One PrepID is defined in the outer level of the request schema while others are defined within tasks/steps. In our use case, PrepID is needed to group the original workflow with its ACDC as Dima is doing in his pages. In that sense, the outer level PrepID seems sufficient to me. Do you agree?

Reply:
If that's the only use case for PrepID within Unified/Dima's page, yes it makes sense.

Reply:
Yes, it will be the only use case. Thanks!